roles/edpm_update_services/tasks/containers.yml

```diff
@@ -0,0 +1,162 @@
+---
+
+- name: Login to container registries if needed
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_podman
+    tasks_from: login.yml
+    apply:
+      become: true
+  tags:
+    - edpm_podman
+    - edpm_update_services
+
+- name: Updates containers for edpm_iscsid role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_iscsid
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_iscsid
+    - edpm_update_services
+  # iscsid is part of the "nova" EDPM service
+  when: '"nova" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_ovn role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_ovn
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_ovn
+    - edpm_update_services
+  when: '"ovn" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_frr role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_frr
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_frr
+    - edpm_update_services
+  when: '"frr" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_ovn_bgp_agent role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_ovn_bgp_agent
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_ovn_bgp_agent
+    - edpm_update_services
+  when: '"ovn-bgp-agent" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_neutron_metadata role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_neutron_metadata
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_neutron_metadata
+    - edpm_update_services
+  when: '"neutron-metadata" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_neutron_ovn role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_neutron_ovn
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_neutron_ovn
+    - edpm_update_services
+  when: '"neutron-ovn" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_multipathd role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_multipathd
+    tasks_from: run.yml
+    apply:
+      become: true
+  tags:
+    - edpm_multipathd
+    - edpm_update_services
+  # multipathd is part of the "nova" EDPM service
+  when: '"nova" in edpm_update_services_running_services'
+
+- name: Apply updates for edpm_nova role
+  ansible.builtin.include_role:
+    name: osp.edpm.edpm_nova
+    tasks_from: update.yml
+  tags:
+    - edpm_nova
+    - edpm_update_services
+  when: '"nova" in edpm_update_services_running_services'
+
+- name: Updates containers for edpm_nova role
```

Comment:
Is there a TODO here to have the nova role have a single entrypoint that has everything in it, rather than requiring an external playbook to know the internals of the role?

Reply:
I think the point of Jesse's comment was that we're including two task lists from the Nova role, and ideally we'd be including just one. Is that correct @odyssey4me ?

Reply:
I think it would be nice if edpm_nova's update.yml included the install.yml, which would mean only update.yml needs to be included here.
that's separate from this PR though.

Reply:
Yes, my intent is to say that the entrypoint should be something like 'update.yml' and then the inclusion of other task files in whatever order should be handled as an implementation detail inside the role, not inside an external playbook. Granted that should be a separate PR, which is why I suggested a TODO. I've created RHOSSTRAT-797 as an Initiative to follow up on this later.

Reply:
The ask is not for edpm_nova install.yml to include update.yml. It's instead that edpm_nova offer a single task file (ideally update.yml) that can be included from edpm_update. Just as the majority of the other roles offer, and not for nova to do its own thing.

edpm_nova update.yml should include the necessary tasks to update nova, which includes any migration tasks, and container restarts, etc. As described in https://github.com/openstack-k8s-operators/dev-docs/blob/main/edpm_update_overview.md